server/src/main/java/org/candlepin/resource/OwnerResource.java

```diff
@@ -342,101 +334,21 @@ public Owner createOwner(Owner owner) {
     public void deleteOwner(@PathParam("owner_key") String ownerKey,
         @QueryParam("revoke") @DefaultValue("true") boolean revoke) {
         Owner owner = findOwner(ownerKey);
-        Event e = eventFactory.ownerDeleted(owner);
+        Event event = eventFactory.ownerDeleted(owner);
 
-        cleanupAndDelete(owner, revoke);
-
-        sink.queueEvent(e);
-    }
-
-    private void cleanupAndDelete(Owner owner, boolean revokeCerts) {
-        log.info("Cleaning up owner: " + owner);
-        List<Consumer> consumers = consumerCurator.listByOwner(owner);
-
-        for (Consumer c : consumers) {
-            log.info("Removing all entitlements for consumer: {}", c);
-
-            poolManager.revokeAllEntitlements(c, revokeCerts);
+        try {
+            ownerManager.cleanupAndDelete(owner, revoke);
         }
-
-        // Actual consumer deletion had to be moved out of
-        // the loop above since all entitlements needed to
-        // be removed before the deletion occured. This is
-        // due to the sourceConsumer that was added to Pool.
-        // Deleting an entitlement may result in the deletion
-        // of a sub pool, which would cause issues.
-        // FIXME  Perhaps this can be handled a little better.
-        for (Consumer consumer : consumers) {
-            // need to check if this has been removed due to a
-            // parent being deleted
-            // TODO: There has to be a more efficient way to do this...
-            log.info("Deleting consumer: {}", consumer);
-            Consumer next = consumerCurator.find(consumer.getId());
-            if (next != null) {
-                consumerCurator.delete(next);
+        catch (PersistenceException e) {
+            if (e.getCause() instanceof ConstraintViolationException) {
+                throw new ConflictException(e.getMessage(), e);
+            }
+            else {
+                throw e;
             }
         }
 
-        for (ActivationKey key : activationKeyCurator
-            .listByOwner(owner)) {
-            log.info("Deleting activation key: {}", key);
-            activationKeyCurator.delete(key);
-        }
-        for (Environment e : owner.getEnvironments()) {
-            log.info("Deleting environment: {}", e.getId());
-            envCurator.delete(e);
-        }
-
-        for (Pool p : poolManager.listPoolsByOwner(owner)) {
-            log.info("Deleting pool: {}", p);
-            poolManager.deletePool(p);
-        }
-
-        cleanupUeberCert(owner);
```

Comment:
@wottop this was only a three line method, and the only instance where it was called, so I replaced the contents in OwnerManager, and removed the method.